sys/net/application_layer/dhcpv6/client.c

```diff
@@ -828,6 +1055,16 @@ static void _request_renew_rebind(uint8_t type)
                     mrd = valid_until;
                 }
             }
+            if (IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA)) {
+                /* calculate MRD from addr_leases */
+                for (unsigned i = 0; i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX; i++) {
+                    const addr_lease_t *lease = &addr_leases[i];
+                    uint32_t valid_until = lease->valid_until / MS_PER_SEC;
```

Comment:
As far as I can tell, you defined the lease time to be in milliseconds. By the multiplications above you risk an overflow and by this division you risk an underflow. Why not keep it seconds as it is provided in the option?

Reply:
I have to admit that I borrowed that code from `dhcpv6_client_conf_prefix` and `dhcpv6_client_prefix_valid_until`. I noticed only now that this conversion is only relevant for the NIB therefore removing the conversion makes a lot of sense. I was wondering, however, if the IA_NA addresses should somehow be integrated into the NIB as well? Or is it only relevant for delegated prefixes?

Reply:
With #16668 and #16679 some clarity about the time units should be in the current code now. I think it makes more sense to borrow from that ;-).